prow/plank/plank.go

```diff
@@ -27,16 +27,18 @@ import (
 
 // NewProwJob initializes a ProwJob out of a ProwJobSpec.
 func NewProwJob(spec kube.ProwJobSpec) kube.ProwJob {
+	jobUUID := uuid.NewV1().String()
 	return kube.ProwJob{
 		APIVersion: "prow.k8s.io/v1",
 		Kind:       "ProwJob",
 		Metadata: kube.ObjectMeta{
-			Name: uuid.NewV1().String(),
+			Name: jobUUID,
 		},
 		Spec: spec,
 		Status: kube.ProwJobStatus{
 			StartTime: time.Now(),
 			State:     kube.TriggeredState,
+			UUID:      jobUUID,
```

Comment:
I agree that we *could* just use `Name` but I'm not sure the coupling is actually the better trade off. I wanted jobs to explicitly have a `UUID` known to them and their child jobs.